### PR TITLE
Add function to get SDK and Lang version

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

I confirm that [`docs.cedarpolicy.com`](https://docs.cedarpolicy.com/) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar language specification.

### CODE REVIEW

**Review by aaronjeline — APPROVED** [review]

**Comment by john-h-kastner-aws** [comment]
If we want this function (as opposed to a function returning the lang version). We should be able to get the SDK [version from env vars](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) to avoid the need for a manual update.
 
> * `CARGO_PKG_VERSION` — The full version of your package.
> * `CARGO_PKG_VERSION_MAJOR` — The major version of your package.
> * `CARGO_PKG_VERSION_MINOR` — The minor version of your package.
> * `CARGO_PKG_VERSION_PATCH` — The patch version of your package.
> * `CARGO_PKG_VERSION_PRE` — The pre-release version of your package.

File: cedar-policy/src/api.rs [hunk 1]
```diff
@@ -59,6 +59,29 @@ use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
...
+pub(crate) mod version {
+    use lazy_static::lazy_static;
+    use semver::Version;
+
+    lazy_static! {
+        // Cedar Rust SDK Semantic Versioning version
+        static ref SDK_VERSION: Version = env!("CARGO_PKG_VERSION").parse().unwrap();
+        // Cedar language version
+        // The patch version field may be unnecessary
+        static ref LANG_VERSION: Version = Version::new(4, 0, 0);
```
Comment (khieta) [inline cedar-policy/src/api.rs:73]: Nit: Cedar lang version only uses MAJOR/MINOR, per https://docs.cedarpolicy.com/other/doc-history.html

Comment (shaobo-he-aws) [reply cedar-policy/src/api.rs:73]: Yeah. The semver crate won't let me to specify optional patch number. Should we make a wrapper out of it?

Comment (khieta) [reply cedar-policy/src/api.rs:73]: If it's not easy to change, then it's fine as-is.

**Review by khieta — COMMENTED** [review]
Can you add a function to the CLI to get the version number?

**Comment by shaobo-he-aws** [comment]
> Can you add a function to the CLI to get the version number?

Yes, working on it.

File: cedar-policy-cli/src/main.rs [hunk 2]
```diff
@@ -53,5 +53,6 @@ fn main() -> CedarExitCode {
         Commands::TranslateSchema(args) => translate_schema(&args),
         Commands::New(args) => new(&args),
         Commands::PartiallyAuthorize(args) => partial_authorize(&args),
+        Commands::LanguageVersion => language_version(),
```
Comment (khieta) [inline cedar-policy-cli/src/main.rs:56]: I think we should have a single `version` command that prints both the language & SDK versions.

Comment (khieta) [reply cedar-policy-cli/src/main.rs:56]: (But aside from that I'm happy with this PR)

Comment (shaobo-he-aws) [reply cedar-policy-cli/src/main.rs:56]: Yeah. Users can get the SDK version just using `--version`.

**Review by khieta — APPROVED** [review]

**Pull request merged**